src/GUI/RevEng.Core/Tables/CSharpModelGeneratorExtend.cs

```diff
@@ -0,0 +1,115 @@
+﻿using Microsoft.EntityFrameworkCore.Diagnostics;
+using Microsoft.EntityFrameworkCore.Metadata;
+using Microsoft.EntityFrameworkCore.Scaffolding;
+using Microsoft.EntityFrameworkCore.Scaffolding.Internal;
+using RevEng.Shared;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+
+namespace RevEng.Core.Tables
+{
+    /// <summary>
+    ///     This is an extension of internal API that supports the Entity Framework Core infrastructure and not subject to
+    ///     the same compatibility standards as public APIs. The base CSharpModelGenerator may be changed or removed without 
+    ///     notice in any release. Updating Entity Framework Core version in project may result application failures.
+    ///     
+    ///     Base class implementation: https://github.com/dotnet/efcore/blob/main/src/EFCore.Design/Scaffolding/Internal/CSharpModelGenerator.cs
+    /// </summary>
+    public class CSharpModelGeneratorExtend : CSharpModelGenerator
+    {
+        /// <summary>
+        ///     This is an extension of internal API that supports the Entity Framework Core infrastructure and not subject to
+        ///     the same compatibility standards as public APIs. The base CSharpModelGenerator may be changed or removed without 
+        ///     notice in any release. Updating Entity Framework Core version in project may result application failures.
+        /// </summary>
+        private ReverseEngineerCommandOptions ReverseEngineerCommandOptions { get; set; }
```

Comment:
Just " bool UseSchemaFolders´"

Reply:
I wonder if this could simply be done here: https://github.com/ErikEJ/EFCorePowerTools/blob/master/src/GUI/RevEng.Core/ReverseEngineerScaffolder.cs#L220 (so no need to override EF Core class)

Reply:
Replaced ReverseEngineerCommandOptions object with UseSchemaFolders property in dependency injection in the latest commit.